docs/testing-plan.md

```diff
@@ -0,0 +1,130 @@
+
+## Re-testing
+If a problem is reported then after a fix is made re-testing of the application needs to be done. Not only for just the initial problem but all tests to make sure new problems did not result due to the change.
+
+If new problems are found see `problem reporting` above else mark the pull request as `Approved` to indicate no new problems were found and the initial problem was solved.
+
+# 4.0 Testing Strategy
+There are many different types of testing stragies including but not limited to:
+- Unit Testing 
+We currently use jest
+
+- System and Integration Testing 
+We are exploring the use of cypress
+
+- Performance and Stress Testing
+
+- User Acceptance Testing
+Our users are often the Docs team or ourselves for user acceptance testing.
+
+- Batch Testing
+
+- Automated Regression Testing
+
+- Beta Testing
+This should take place when we are reviewing the PRs and spin up the application locally.
+
+# 5.0 Hardware Requirements
+Currently we only support spinning up development enviroments on Mac books.
+
+# 6.0 Features to Be Tested
+Note as this application is constnatly being iterated on this features to be tested list should always be reviewed and edited based on features added / removed or updated.
+
+These descriptions are at a high level and the best indicator when testing development is to view what the application looks like in productions.
+
+Remember to test all features in both English and Japanese version of the Docs site.
+
+## Accessibility testing
+- Can you search in algolia?
+- Are the tabs in search toggleable?
+- Can you tab to `Skip to Content`?
+
+## Header
+- CircleCi logo that when clicked takes you to docs homepage
+- Search bar (See search bar below more details)
+- Globe icon that opens to langauge selector (See language selector below for more details)
+- Link to circleci.com
+- Button to `Go to application` or `Sign up`
+
+## Search bar
+- When typing inside of search bar then results a results page should appear. The url should now contain a query string of the word used. 3 tabs should appear
+1. Documention - results found in /docs
+2. Orbs - Results found in the developer/orb site
+3. Convenience Images - Results found in developer/orb
+- Pressing cancel should close search bar results 
+- Pressing `X` should clear search results and query string from url
+
+## Side nav
+- Contains a list to all different resource available on the Docs site
+- There should be top level sections that toggle between open and closed when clicked on revealing child lists 
+- The side nav should ensure the parent category is opened and the child side page header is highlighted in the list for the associated page that the user is currently viewing
+- When an item is clicked on the side nav it should navigate the user to the associated page, and highlight the item should be highlighted
+
+## TOC
+- TOC may not appear on all pages
+- TOC contains a list of headers that are on the page and helpful resources
```

Comment:
TOC and Helpful resources are two different things. Let's not mix both in the same section please :)